modules/claims_api/app/controllers/claims_api/v0/forms/disability_compensation_controller.rb

```diff
@@ -42,14 +43,19 @@ def upload_supporting_documents
         end
 
         def validate_form_526
-          validate_form526(form_attributes)
-          render json: { data: 'success' }
+          valid_service = EVSS::DisabilityCompensationForm::ServiceAllClaim.new(auth_headers)
+          validation_result = valid_service.validate_form526(form_attributes.to_json)
+          render json: validation_result
+        rescue EVSS::ErrorMiddleware::EVSSError => e
+          render json: { errors: format_errors(e.details) }, status: :unprocessable_entity
         end
 
         private
 
-        def validate_form526(form_data)
-          service(auth_headers).validate_form526(form_data)
+        def format_errors(errors)
+          errors.map do |error|
+            { status: 422, detail: "#{error['key']} #{error['detail']}", source: nil }
```

Comment:
is there anypoint to doing `source: nil` could it be `source: error['key']`?

Reply:
That or possibly external? What do you think?